pkg/services/ngalert/remote/external_alertmanager.go

```diff
@@ -12,13 +12,14 @@ import (
 	"github.com/grafana/grafana/pkg/infra/log"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
 	amclient "github.com/prometheus/alertmanager/api/v2/client"
 	amalert "github.com/prometheus/alertmanager/api/v2/client/alert"
 	amalertgroup "github.com/prometheus/alertmanager/api/v2/client/alertgroup"
 	amsilence "github.com/prometheus/alertmanager/api/v2/client/silence"
 )
 
-type externalAlertmanager struct {
+type ExternalAlertmanager struct {
```

Comment:
Why are we exporting the `externalAlertmanager` struct? There's a public factory function and all the methods we need to use are also public. We only care about this struct implementing the `Alertmanager` interface.
I would only export `ExternalAlertmanagerConfig`

Reply:
My IDE threw a warning (and it typically is a code smell) that I had a public method returning a private struct.